src/main/common/kalman.c

```diff
@@ -116,17 +116,15 @@ FAST_CODE float kalman_process(kalman_t* kalmanState, float input, float target)
   //update last state
   kalmanState->lastX = kalmanState->x;
 
+  if (kalmanState->lastX != 0.0f) {
```

Comment:
guess it's fine but try to have the most common case first - for performance reasons :)

Reply:
Yea that isn't a very common case at all, but how would you suggest I change it, and still maintain its functionality?

Reply:
try to use `==` whenever possible and match the most probable case first. guess it's fine for now

Reply:
I mean in this code this is added just to prevent a divide by 0 error. This should be good to merge.

Reply:
i wasnt saying it is not. at the time of my previous writing the status checks were still running. you should be open to suggestions though and not just pushing for approval.